### PR TITLE
Remove check that enforces slack oauthProviders are set

### PR DESCRIPTION
**Pull request by EdwardDowling**
This check needs to be removed in advance of other plugins being self hosted.
Also sets hostedplugins to always be enabled

Part of [30137](https://github.com/gravitational/teleport/issues/30137)

### CODE REVIEW

**Review by ibeckermayer — COMMENTED** [review]
Code looks fine to me but perhaps @justinas should double check that this won't break anything.

**Review by justinas — APPROVED** [review]
Should be okay

**Review by strideynet — APPROVED** [review]

**Comment by EdwardDowling** [comment]
> Should be okay

Do the changes that always enable hosted plugins look good to you as well?

**Review by ibeckermayer — APPROVED** [review]

File: lib/config/configuration.go [hunk 1]
```diff
@@ -850,12 +850,10 @@ func applyAuthConfig(fc *FileConfig, cfg *servicecfg.Config) error {
 
 	cfg.Auth.LoadAllCAs = fc.Auth.LoadAllCAs
 
-	if fc.Auth.HostedPlugins.Enabled {
-		cfg.Auth.HostedPlugins.Enabled = true
-		cfg.Auth.HostedPlugins.OAuthProviders, err = fc.Auth.HostedPlugins.OAuthProviders.Parse()
-		if err != nil {
-			return trace.Wrap(err)
-		}
+	cfg.Auth.HostedPlugins.Enabled = true
```
Comment (r0mant) [inline lib/config/configuration.go:853]: Let's add a comment here explaining why we always set it to true.

**Pull request merged**